activerecord/lib/active_record/insert_all.rb

```diff
@@ -198,7 +198,19 @@ def updatable_columns
         def touch_model_timestamps_unless(&block)
           model.timestamp_attributes_for_update_in_model.filter_map do |column_name|
             if touch_timestamp_attribute?(column_name)
-              "#{column_name}=(CASE WHEN (#{updatable_columns.map(&block).join(" AND ")}) THEN #{model.quoted_table_name}.#{column_name} ELSE CURRENT_TIMESTAMP END),"
+              # TODO: Make sure column_name is a string, not a symbol, to match column.name
+              # TODO: Should this use the undocumented model.columns_hash to lookup without iteration?
+              #       i.e precision = model.columns_hash.fetch(column_name).precision
+              precision = model.columns.find { |column| column.name == column_name }.precision
+              precision ||= 0
+              # TODO: Do we even need to calculate the precision, or can we just use the max precision every time
+              #       (i.e. hardcode CURRENT_TIMESTAMP(6) or whatever the max precision is, allowed by this DB adapter)
+              # NOTE: Not all DBs allow parens after CURRENT_TIMESTAMP.
+              #         e.g. - MySQL considers CURRENT_TIMESTAMP, CURRENT_TIMESTAMP(), & CURRENT_TIMESTAMP(0) identical,
+              #                and allows up to CURRENT_TIMESTAMP(6)
+              #              - SQLite only accepts CURRENT_TIMESTAMP, and does not allow the use of parentheses at all
+              current_timestamp = precision.zero? ? "CURRENT_TIMESTAMP" : "CURRENT_TIMESTAMP(#{precision})"
```

Comment:
It would seem I have opened a can of worms, since there are weird scenarios like SQLite supporting fractional timestamps, but not accepting `CURRENT_TIMESTAMP(6)`.

**Is there a reason we rely on `CURRENT_TIMESTAMP` in the first place, instead of just passing in a timestamp derived from `Time.now`?**

Some quick googling finds things like 

> [The 'now' argument to date and time functions always returns exactly the same value for multiple invocations within the same sqlite3_step() call.](https://www.sqlite.org/lang_datefunc.html)

so I don't think there's any difference in the timestamps even if we perform a large bulk insert (assuming this kind of behaviour is consistent across other databases).

As far as I can tell, the only difference we'd see between using `CURRENT_TIMESTAMP` and injecting `Time.now` would be the small difference in time between the timestamp generated by the database, and the timestamp generated by us before the database receives our query, but that is likely inconsequential.